### ISSUE TITLE
Amazon Prime - Baldurs Gate Enhanced Edition - JSON error when trying to install

### ISSUE THREAD

**Issue by eris667** [lutris/lutris#4785]
Nobara Linux 37
Lutris 0.5.13 (package == not flatpak)
Amazon Prime - Baldurs Gate Enhanced Edition - JSON error when trying to install
**Invalid JSON response from Amazon APIs**

Lutris -d (pastebin)
https://pastebin.com/T5FGgy2g

Image
https://media.discordapp.net/attachments/538903130704838656/1088220813599195186/image.png?width=882&height=425

**Comment by eris667** [comment]
Updated Error 04.07.23
Nobara Linux 37
Lutris 0.5.13 (package == not flatpak)
Amazon Prime - Baldurs Gate Enhanced Edition - error when trying to install
('Unable to get the patches of game, please check your Amazon credentials and internet connectivity', '12c3557c-0d05-5275-aef8-4f3b98bedaca')

Lutris -d (pastebin)
https://pastebin.com/WCkQpmxn

Image
https://media.discordapp.net/attachments/538903130704838656/1093878032198025287/image.png?width=533&height=504

**Comment by danieljohnson2** [comment]
Did you check your credentials and internet connectivity?

**Comment by eris667** [comment]
> Did you check your credentials and internet connectivity?

I was able to install 3 other games from Amazon Prime with no issues, the only game giving an error is this one (so internet and credentials should be ok?)

**Comment by danieljohnson2** [comment]
That rather indicates it isn't your connection, yes.

You've tried 0.5.13 beta, but there have been changes to the Amazon patch stuff since then. Could you download the master branch and try it?

No need to install; just download the latest and run the bin/lutris script within. Try to install the game and see if it does any better.

**Comment by eris667** [comment]
ok so I went here:
https://github.com/lutris/lutris
Code - Download as Zip - lutris-master.zip
Extracted
/home/eris667/Downloads/lutris-master/bin
lutris - Run as a Program
Lutris opens
BG - Install - Back to the original JSON error
_Invalid JSON response from Amazon APIs_

**Comment by danieljohnson2** [comment]
OK, I think I see what Amazon has done wrong with this JSON. There is an extra stray comma before the "Main" entry. I can write some RegEx to strip that out.

Please try [my branch](https://github.com/lutris/lutris/tree/dj/all_the_json).

Go to that link, and download Lutris from there as before; just run 'bin/lutris' and see if that allows you to install your game.

**Comment by eris667** [comment]
Installing :100:

**Comment by danieljohnson2** [comment]
Do you mean it worked? If so I'll merge that into master.

**Comment by eris667** [comment]
I may need to `lutris -d `and get a log, trying again to see if it hangs up on the same files in the download process

**Comment by eris667** [comment]
successfully installed and launched - tyvm!

**Comment by eris667** [comment]
successfully installed and launched - tyvm!

**Comment by danieljohnson2** [comment]
Very good; I've merged this into master.

It has been suggested to me on Discord that this might be YAML made to look like JSON. Your example does indeed parse as YAML, but there was already code fudging this 'JSON' and it's not clear how that fudge would help YAML pass for JSON.

Since I have no Amazon account and can't test it, I won't proceed further with the YAML idea- at least not until Amazon's "JSON" fails us again.

**Comment by danieljohnson2** [comment]
I've got a volunteer to test it, so I'm trying a YAML based fix for this. If you have the time for this, you might wish to retest this with the very latest master- it should work just the same as the last fix.

If this works it should be much safer than munging the response with regular expressions.

**Comment by eris667** [comment]
Uninstalled BG, Downloaded latest lutris-master, launched (via extracted /bin)
BG Install progressed to download UI with no errors
downloaded BG (hundreds of small files)
`2023-04-08 07:03:40,167: Download failed: HTTPSConnectionPool(host='client.legacy-app.games.a2z.com', port=443): Read timed out. (read timeout=30)`

restarted lutris-master, restarted download with no issues (it takes ~10m on 1g fiber, game folder size is ~4g)
```
2023-04-08 07:26:48,140: All files available
2023-04-08 07:26:48,140: All files are available, continuing install
2023-04-08 07:26:48,255: Creating a win64 prefix in /home/eris667/Games/amazon/baldurs-gate-enhanced-edition
```
_successfully installed and launched - tyvm!_

**Comment by danieljohnson2** [comment]
Very good. It seems like Amazon has dolled up their YAML to almost, but not quite, look like JSON. Very helpful.

Thanks for your help.

**Comment by IsenMike** [comment]
I'm currently getting the same 'Unable to get the patches of game, please check your Amazon credentials and internet connectivity' error for Cultist Simulator.  No issue with credentials or connectivity; I'm able to download other games without issue, but attempting to install Cultist Simulator consistently gives this error.

Apologies for lack of debug text; I'm using Steam Deck, have installed Lutris via the Discovery app, and when I try to run 'lutris -d' or 'lutris --submit-issue' from the terminal it just gives me 'command not found' for whatever reason.  Happy to provide the debug text if you can help me figure that out; in the meantime, here's a screenshot of the error, if that helps at all: https://i.imgur.com/AQImwTy.png

**Comment by eris667** [comment]
@IsenMike you posted on a unrelated closed issue, for better visibility you should make a new post dedicated to your issue

**Comment by IsenMike** [comment]
> @IsenMike you posted on a unrelated closed issue, for better visibility you should make a new post dedicated to your issue here: https://github.com/lutris/lutris/issues

Thanks.  I worked out how to get the debug logging to work and created a new issue: https://github.com/lutris/lutris/issues/4845
